src/main/java/com/powsybl/openloadflow/adm/AdmittanceMatrix.java

```diff
@@ -0,0 +1,94 @@
+/**
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+package com.powsybl.openloadflow.adm;
+
+import com.powsybl.math.matrix.DenseMatrix;
+import com.powsybl.math.matrix.LUDecomposition;
+import com.powsybl.math.matrix.Matrix;
+import com.powsybl.math.matrix.MatrixFactory;
+import com.powsybl.openloadflow.equations.EquationSystem;
+import com.powsybl.openloadflow.equations.JacobianMatrix;
+import com.powsybl.openloadflow.network.LfBus;
+import org.apache.commons.math3.complex.Complex;
+
+import java.util.Objects;
+
+/**
+ * @author Geoffroy Jamgotchian <geoffroy.jamgotchian at rte-france.com>
+ */
+public class AdmittanceMatrix implements AutoCloseable {
+
+    private final EquationSystem<AdmittanceVariableType, AdmittanceEquationType> equationSystem;
+
+    private final Matrix matrix;
+
+    private LUDecomposition lu;
+
+    private DenseMatrix e; // impedance extractor
+
+    public AdmittanceMatrix(EquationSystem<AdmittanceVariableType, AdmittanceEquationType> equationSystem, Matrix matrix) {
+        this.equationSystem = Objects.requireNonNull(equationSystem);
+        this.matrix = Objects.requireNonNull(matrix);
+    }
+
+    public static AdmittanceMatrix create(AdmittanceEquationSystem ySystem, MatrixFactory matrixFactory) {
+        Objects.requireNonNull(matrixFactory);
+        try (var j = new JacobianMatrix<>(ySystem.getEquationSystem(), matrixFactory)) {
+            return new AdmittanceMatrix(ySystem.getEquationSystem(), j.getMatrix());
```

Comment:
Wouldn't it be cleaner to inherit from JacobianMatrix ?
Here if a future evolution of JacobianMatrix closes the Matrix object the could will fail because it assumes today that the matrix is still alive.

Reply:
Admittance matrix has nothing to do with Jacobian matrix even if we use it to compute admittance.

Reply:
Well - you reused the matrix factory part. And steal the matrix of a closable - in addition a matrix with a DirectBuffer in it. That works today, and in your code structure but is not really a reusable object. Not a big deal and not reason to disapprove. But I can't stay silent when I see this :-)